src/components/chat/hooks/useScrollManagement.ts

```
@@ -95,6 +111,53 @@ export function useScrollManagement({
+  // overwrite a real position with scrollTop=0.
+  const lastKnownScrollRef = useRef({
+    sessionId: activeSessionId ?? '',
+    scrollTop: 0,
+    isFollowingTail: true,
+  })
+  const prevSessionIdRef = useRef<string | null | undefined>(activeSessionId)
+  // When true, the next content-ready layout pass should apply a saved
+  // non-tail scrollTop instead of pinning to the bottom.
+  const pendingRestoreRef = useRef(false)
+  const contentReadyRef = useRef(contentReady)
+  contentReadyRef.current = contentReady
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/components/chat/hooks/useScrollManagement.ts:127:3
  125 |   const pendingRestoreRef = useRef(false)
  126 |   const contentReadyRef = useRef(contentReady)
> 127 |   contentReadyRef.current = contentReady
      |   ^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  128 |
  129 |   const persistCurrentSessionScroll = useCallback(
  130 |     (sessionId: string | null | undefined) => {

(react-hooks/refs)